lib/DataHarmonizer.js

```diff
@@ -1967,70 +1985,55 @@ class DataHarmonizer {
         let valid = false;
 
         if (!cellVal) {
-          if (field.required) msg = 'Required cells cannot be empty';
-          else valid = true;
+          if (field.required) {
+            msg = 'Required cells cannot be empty';
+          } else {
+            valid = true;
+          }
         }
 
         // If not an empty field, check its contents against field datatype AND/OR other kind of range
         else {
-          let parsedInt;
-          let parsedFloat;
-          let parsedDec;
+          // parseDatatype will return undefined for any value that doesn't
+          // match the pattern of the given datatype
+          const parsed = parseDatatype(cellVal, datatype);
+          valid = parsed !== undefined;
 
           switch (datatype) {
             case 'xsd:integer':
-              parsedInt = parseInt(cellVal, 10);
-              valid = !isNaN(cellVal);
-              valid &= parsedInt.toString() === cellVal;
-              valid &= testNumericRange(parsedInt, field);
+              valid &&= testNumericRange(parsed, field);
               break;
 
             case 'xsd:nonNegativeInteger':
-              parsedInt = parseInt(cellVal, 10);
-              valid = !isNaN(cellVal) && parsedInt >= 0;
-              valid &= parsedInt.toString() === cellVal;
-              valid &= testNumericRange(parsedInt, field);
+              valid &&= parsed >= 0;
+              valid &&= testNumericRange(parsed, field);
               break;
 
             case 'xsd:float':
-              parsedFloat = parseFloat(cellVal);
-              valid = !isNaN(cellVal) && parsedFloat == cellVal;
-              valid &= testNumericRange(parsedFloat, field);
+              valid &&= testNumericRange(parsed, field);
               break;
 
             case 'xsd:double':
               // NEED DOUBLE RANGE VALIDATION
-              parsedFloat = parseFloat(cellVal);
               //valid = !isNaN(cellVal) && regexDouble.test(cellVal);
-              valid &= !isNaN(cellVal) && testNumericRange(parsedFloat, field);
+              valid &&= testNumericRange(parsed, field);
               break;
 
             case 'xsd:decimal':
-              parsedDec = parseFloat(cellVal);
-              valid = !isNaN(cellVal) && regexDecimal.test(cellVal);
-              valid &= testNumericRange(parsedDec, field);
+              valid &&= testNumericRange(parsed, field);
               break;
 
-            // XML Boolean lexical space accepts true, false, and also 1
-            // (for true) and 0 (for false).
             case 'xsd:boolean':
-              valid =
```

Comment:
So a field, if parsed as boolean, then doesn't need its value checked (already done).  Ok!

Reply:
Exactly, the check for `undefined` a few lines up is sufficient for `xsd:boolean`